RELEASE.md

```diff
@@ -75,24 +75,29 @@ you can skip the dependency update or only update select dependencies. In such a
 case, you have to create an issue or pull request in the GitHub project for
 later follow-up.
 
-### Prepare your release
+### 1. Prepare your release
 
-For a patch release, work in the branch of the minor release you want to patch.
+Create a new release branch.
```

Comment:
That's not true. Only a new minor or major release requires a now release branch. For a patch release, you must not create a new release branch.